### PR TITLE
VoIP Screen Sharing Permission

### PR DESCRIPTION
**Pull request by onurays**
Initial PR for screen sharing during a call.

- [x] Add screen sharing button into call screen
- [x] Show permission dialog
- [x] Start a foreground service with a sticky notification for Android Q and later
- [x] Toggle start/stop screen sharing button

| Share Screen Button | Permission Dialog | Stop Sharing Screen |
| ------------------- | ----------------- | -------------------- |
| screen_sharing_1 | screen_sharing_2 | screen_sharing_3

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/call/VectorCallViewEvents.kt [hunk 1]
```diff
@@ -30,6 +30,8 @@ sealed class VectorCallViewEvents : VectorViewEvents {
     object ShowDialPad : VectorCallViewEvents()
     object ShowCallTransferScreen : VectorCallViewEvents()
     object FailToTransfer : VectorCallViewEvents()
+    object ShowScreenSharingPermissionDialog : VectorCallViewEvents()
+    object StopScreenSharingService : VectorCallViewEvents()
```
Comment (ericdecanini) [inline vector/src/main/java/im/vector/app/features/call/VectorCallViewEvents.kt:34]: Optional: This is a good opportunity to get rid of the commented code below. They're nearly 2 years old, I don't think we'll need it

File: vector/src/main/java/im/vector/app/features/call/webrtc/ScreenCaptureService.kt [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+class ScreenCaptureService : VectorService() {
+
+    @Inject lateinit var notificationUtils: NotificationUtils
+    private val binder = LocalBinder()
+
+    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
+        // Show a sticky notification
+        val notificationId = System.currentTimeMillis().toInt()
+        val notification = notificationUtils.buildScreenSharingNotification()
+        startForeground(notificationId, notification)
```
Comment (ericdecanini) [inline vector/src/main/java/im/vector/app/features/call/webrtc/ScreenCaptureService.kt:37]: Optional: extracting this code into a `private fun showStickyNotification()` could get rid of the need for a comment ;)

**Review by ericdecanini — APPROVED** [review]
LGTM! Left a couple of optional suggestions

File: vector/src/main/java/im/vector/app/features/call/VectorCallViewEvents.kt (hunk 1 above)
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/call/VectorCallViewEvents.kt:35]: Right :) done.

File: vector/src/main/java/im/vector/app/features/call/webrtc/ScreenCaptureService.kt (hunk 2 above)
Comment (onurays) [reply vector/src/main/java/im/vector/app/features/call/webrtc/ScreenCaptureService.kt]: Nice idea, done.

**Pull request merged**